metrics-operator/controllers/common/providers/prometheus/common_test.go

```diff
@@ -119,7 +119,8 @@ func Test_GetRoundtripper(t *testing.T) {
 		name      string
 		provider  metricsapi.KeptnMetricsProvider
 		k8sClient client.Client
-		want      http.RoundTripper
+		wantUser  string
```

Comment:
what is the reasoning behind this change? the function returns a roundtripper, therefore I would expect that instead of user and password

Reply:
can you elaborate this little? @odubajDT

Reply:
can you please elaborate on this? @odubajDT

Reply:
my question is why you are expecting a user and password if the function you are testing is returning a roundtripper, it should in my opinion stay as it is, unless there is a reason for the change

Reply:
The thing is for the unit test without user and password it was failing so when I used user and password also it passed so I thought this way works when using user and password

Reply:
yes the test is passing because now you are not checking the roundTripper correctly, please revert this and try to stick the the concept that is present unless there is a technical reason why it should be done differently

Reply:
please try to look at all the other tests present for the same function. It should be very similar

Reply:
one question:  that if I want to use BasicAuthRoundTripper in the test then now I use it config.BasicAuthRoundTripper but it throws error as there is no BasicAuthRoundTripper  in the config package 
@odubajDT

Reply:
Hey @odubajDT 
The thing is that when i added **InsecureSkipTlsVerify: true** to the provider's spec and updated the want field to use a custom **http.Transport** with **TLSClientConfig set to InsecureSkipVerify: true.** the test fails 
**getRoundtripper() got = &{myuser mytoken   0x140005afb00}, want &{myuser mytoken   0x140001d7980}** failing because the comparison of RoundTripper instances is not working with reflect.DeepEqual. 

so instead of comparing entire RoundTripper structs I check specific properties like wantUser, wantPass, and wantTLS instead of a complete want RoundTripper.

Reply:
@odubajDT one way is that by creating mock HTTP server to capture and verify the actual request sent by the RoundTripper. Instead of checking the request directly.

Reply:
I guess you need to adapt the expected outcome of the function, since you just added a parameter to the config, which has an impact on the resulting object

Reply:
Can I create a new function the **common_test** file?

Reply:
what you should do is to set/unset the parameter in the metricProvider and expect it to be set/unset in the resulting roundtripper in the test. Not many changes are needed and I don't think another test function is needed here. You just need to duplicate and slightly adapt the test cases that are present

Reply:
Hey @odubajDT 
I was modifying test cases but the `config.BasicAuthRoundTripper` type is not directly accessible and when i changed it to `*http.Transport` the test fails as GetRoundTripper function is returning `*config.basicAuthRoundTripper` instead of a `*http.Transport`